packages/core/src/scuttle.js

```diff
@@ -0,0 +1,185 @@
+/**
+ * @typedef {object} ScuttleOpts
+ * @property {boolean} enabled - Whether scuttling is enabled or not.
+ * @property {Array<string|RegExp>} exceptions - List of properties to exclude from scuttling.
+ * @property {string} scuttlerName - Name of the scuttler function to use.
+ * @property {(globalRef: object, scuttle: Function) => void} scuttlerFunc - Scuttler function to use.
+ */
+
+/**
+ * @typedef {object} GlobalRef
+ * @property {object} [globalThis] - Reference to the global object.
+ */
+
+const {
+  assign,
+  getOwnPropertyNames,
+  getOwnPropertyDescriptor,
+  create,
+  defineProperty,
+} = Object
+
+/**
+ * Applies scuttling, with the default set of options, including using Snow if passed in as scuttlerFunc.
+ * cSuttle globalThis right after we used it to create the root package compartment
+ *
+ * @param {GlobalRef} globalRef - Reference to the global object.
+ * @param {ScuttleOpts} opts - Scuttling options.
+ */
+function applyDefaultScuttling(globalRef, opts) {
```

Comment:
If we are willing to change the signature here, I like it when "options" are all optional. 😄